.github/workflows/publish_pypi_package.yml

```
@@ -0,0 +1,121 @@
+            exit 1
+          fi
+
+      - id: determine-ref
+        name: "Determine github commit REF to be used"
+        run: |
+          REF=""
+          if [[ "" != "${{ github.event.client_payload.reference }}" ]]; then
+            REF="${{ github.event.client_payload.reference }}"
+            echo "Setting reference from client_payload: $REF"
+          elif [[ "" != "${{ github.event.inputs.ref }}" ]]; then
+            REF="${{ github.event.inputs.ref }}"
+            echo "Setting ref from github.event.inputs: $REF"
+          else
+            echo "::notice::No REF Provided."
+          fi
+          echo "::set-output name=ref::$REF"
+
+  validate-tag:
+    needs: determine-tag-and-ref
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: "Validate Tag"
+        uses: tetrascience/ts-action-validate-and-tag@main
+        with:
+          tag: ${{ needs.determine-tag-and-ref.outputs.tag }}
+          ref: ${{ needs.determine-tag-and-ref.outputs.ref }}
+
+  publish-package:
+    needs: validate-tag
+    runs-on: ubuntu-latest
+    environment: PROD
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-python@v3
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    # Temporary - to be removed once ts-sdk uses poetry
+    - name: Build ts-sdk package
+      if: hashFiles('poetry.lock') == ''    
+      run: |
+        pip install --upgrade pip setuptools wheel
+        python3 setup.py bdist_wheel
+        pip install ./dist/*.whl
+        ts-sdk --help
+        ts-sdk init -o org -p test-protocol -t ts-slug -f ./new-protocol
+        pushd .
+        cd ./new-protocol/task-script
+        pipenv install --dev ../../dist/*.whl
+        pipenv install --dev
+        pipenv run python -m pytest -s
+        popd
+
+    - name: Build package
+      if: hashFiles('poetry.lock') != ''
```

Should we raise an error if there's no `poetry.lock` file?

---

No need, the workflow will fail by itself

---

We also have a limited number of packages we are publishing

---

> No need, the workflow will fail by itself

I don't wanna hold up the PR too much, but as the Workflow currently stands the error message would not be very clear. I recommend adding a test that raises an error like, "Only publishing from `poetry` is implemented. No poetry packaging metadata found."

---

From a Slack conversation, it looks like `setup.py` publishing is handled by the `pypa/gh-action-pypi-publish` below. Approved.